edit/edcore/insert.go

```diff
@@ -210,35 +212,47 @@ func (ed *editor) moveDotRight() {
 	ed.dot += w
 }
 
-func (ed *editor) moveDotLeftWord() {
+func (ed *editor) moveDotLeftSepFunc(fn func(rune) bool) {
 	if ed.dot == 0 {
 		return
 	}
 	space := strings.LastIndexFunc(
-		strings.TrimRightFunc(ed.buffer[:ed.dot], unicode.IsSpace),
-		unicode.IsSpace) + 1
+		strings.TrimRightFunc(ed.buffer[:ed.dot], fn),
+		fn) + 1
 	ed.dot = space
 }
 
-func (ed *editor) moveDotRightWord() {
-	// Move to first space
-	p := strings.IndexFunc(ed.buffer[ed.dot:], unicode.IsSpace)
+func (ed *editor) moveDotRightSepFunc(fn func(rune) bool) {
+	// Move to first matching rune
+	p := strings.IndexFunc(ed.buffer[ed.dot:], fn)
 	if p == -1 {
 		ed.dot = len(ed.buffer)
 		return
 	}
 	ed.dot += p
-	// Move to first nonspace
-	p = strings.IndexFunc(ed.buffer[ed.dot:], notSpace)
+	// Move to first non-matching rune
+	p = strings.IndexFunc(ed.buffer[ed.dot:], func(r rune) bool { return !fn(r) })
 	if p == -1 {
 		ed.dot = len(ed.buffer)
 		return
 	}
 	ed.dot += p
 }
 
-func notSpace(r rune) bool {
-	return !unicode.IsSpace(r)
+func (ed *editor) moveDotLeftWord() {
+	ed.moveDotLeftSepFunc(unicode.IsSpace)
+}
+
+func (ed *editor) moveDotRightWord() {
+	ed.moveDotRightSepFunc(unicode.IsSpace)
+}
+
+func (ed *editor) moveDotLeftSmallWord() {
+	ed.moveDotLeftSepFunc(isAlnum)
```

Comment:
Instead of passing `isAlnum`, pass the negation of `isAlnum`, i.e.

```go
ed.moveDotLeftSepFunc(func(r rune) { return !isAlnum(r) })
```

Ditto below